utils/common/poolallocator.cpp

```diff
@@ -65,32 +65,18 @@ void PoolAllocator::newBlock()
         nextAlloc = mem.front().get();
 }
 
-void* PoolAllocator::allocate(uint64_t size)
+void * PoolAllocator::allocOOB(uint64_t size)
 {
-    void* ret;
+    OOBMemInfo memInfo;
 
-    if (size > allocSize)
-    {
-        OOBMemInfo memInfo;
-
-        memUsage += size;
-        memInfo.mem.reset(new uint8_t[size]);
-        memInfo.size = size;
-        ret = (void*) memInfo.mem.get();
-        oob[ret] = memInfo;
-        return ret;
-    }
-
-    if (size > capacityRemaining)
-        newBlock();
-
-    ret = (void*) nextAlloc;
-    nextAlloc += size;
-    capacityRemaining -= size;
     memUsage += size;
+    memInfo.mem.reset(new uint8_t[size]);
```

Comment:
What happens if new fails? Do we have try-except on top of this?

Reply:
In this case, if nothing else then threadpool or BPPSeeder will catch it.  I don't remember off the top of my head if/how the query is aborted at that point.